soh/soh/Enhancements/AssignableTunicsAndBoots.cpp

```diff
@@ -0,0 +1,108 @@
+#include "soh/Enhancements/game-interactor/GameInteractor_Hooks.h"
+#include "soh/ShipInit.hpp"
+
+extern "C" {
+#include "macros.h"
+#include "variables.h"
+
+extern s32 Player_GetItemOnButton(PlayState*, s32);
+extern void Inventory_ChangeEquipment(s16, u16);
+extern void Player_SetEquipmentData(PlayState*, Player*);
+extern void func_808328EC(Player*, u16);
+}
+
+static u16 sItemButtons[] = { BTN_B, BTN_CLEFT, BTN_CDOWN, BTN_CRIGHT, BTN_DUP, BTN_DDOWN, BTN_DLEFT, BTN_DRIGHT };
+
+void UseTunicBoots(Player* player, PlayState* play, Input* input) {
+    // Boots and tunics equip despite state
+    if (
+        player->stateFlags1 & (PLAYER_STATE1_INPUT_DISABLED | PLAYER_STATE1_IN_ITEM_CS | PLAYER_STATE1_IN_CUTSCENE | PLAYER_STATE1_TALKING | PLAYER_STATE1_DEAD) ||
+        player->stateFlags2 & PLAYER_STATE2_OCARINA_PLAYING
+    ) {
+        return;
+    }
+
+    s32 i;
+    for (i = 0; i < ARRAY_COUNT(sItemButtons); i++) {
+        if (CHECK_BTN_ALL(input->press.button, sItemButtons[i])) {
+            break;
+        }
+    }
```

Comment:
What is this loop for?

Reply:
it's just copied from what was in `z_player` before. `i` is defined outside of the loop and used in `Player_GetItemOnButton` below.

i'll clean it up to make this easier to read

Reply:
changed to https://github.com/HarbourMasters/Shipwright/pull/4978/commits/a15124173380d956bc70da9fd26ba91ba238133d